src/angles.jl

```diff
@@ -27,16 +24,5 @@ function ∠(A::P, B::P, C::P) where {P<:Point{2}}
   cross = BA × BC
   inner = BA ⋅ BC
 
-  R = norm(cross) / (norm(BA) * norm(BC))
-
-  # Table 1 from Balbes, R. and Siegel, J. 1990.
-  if cross ≥ 0 && inner ≥ 0
-    asin(R)
-  elseif cross ≥ 0 && inner < 0
-    π - asin(R)
-  elseif cross < 0 && inner ≥ 0
-    -asin(R)
-  elseif cross < 0 && inner < 0
-    asin(R) - π
-  end
+  atan(cross, inner)
```

Comment:
Can you comment on how this atan(cross, inner) is equivalent to the previous code? I've seen [this method](https://stackoverflow.com/a/21484228) and was wondering how it relates to the above method.

Reply:
`atan` allows you to retrieve a signed angle from its sine and cosine. Here, `cross` and `inner` are the sine and cosine respectively, up to a factor of `norm(BA) * norm(BC)`. This factor is not important however, because `atan` only uses the ratio between the sine and cosine, as well as the sign of both values. The common factor is positive, so we can safely pass in the unscaled values.
Note also that `cross` is a signed value in 2D, but a vector in 3D. Therefore, for the 3D case, this would require more work: we would need to take the norm of the cross product and figure out the sign from BA and BC.

In the method you linked, the angle of both vectors from the origin is computed, and then subtracted. This will only work in 2D as well, in 3D you would have to compute a pointed plane containing both vectors and compute their angles with respect to an arbitrary origin within the plane.

In terms of speed:
```julia
julia> @benchmark ∠(Point(1,0), Point(0,0), Point(0,1)) # first method
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     3.709 ns (0.00% GC)
  median time:      3.931 ns (0.00% GC)
  mean time:        3.997 ns (0.00% GC)
  maximum time:     19.856 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark ∠(Point(1,0), Point(0,0), Point(0,1)) # second method
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     3.122 ns (0.00% GC)
  median time:      3.364 ns (0.00% GC)
  mean time:        3.570 ns (0.00% GC)
  maximum time:     24.943 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```

Both methods therefore seem equivalent.

Reply:
Amazing clarification as usual @serenity4 ❤️